File: automation-test-selenium/src/main/java/org/suntower/pages/core/StaffShellPage.java
package org.suntower.pages.core;

import org.openqa.selenium.WebDriver;
import org.suntower.core.BasePage;

public class StaffShellPage extends BasePage {
  protected String path;

  public StaffShellPage(WebDriver driver) {
    super(driver);
  }

  public void open() {
    if (path == null || path.isBlank()) {
      throw new IllegalStateException("This staff page does not define a path.");
    }
    visit(path);
  }

  public void openDashboard() {
    visit("/staff/dashboard");
  }

  public void goToBuildings() {
    visit("/staff/buildings");
  }

  public void goToCustomers() {
    visit("/staff/customers");
  }

  public void goToContracts() {
    visit("/staff/contracts");
  }

  public void goToInvoices() {
    visit("/staff/invoices");
  }

  public void goToProfile() {
    visit("/staff/profile");
  }
}

File: automation-test-playwright/pages/core/StaffShellPage.ts
import { BasePage } from "./BasePage";

export class StaffShellPage extends BasePage {
  protected readonly path?: string;

  /**
   * Opens the page.
   */
  async open(): Promise<void> {
    if (!this.path) {
      throw new Error("This staff page does not define a path.");
    }

    await this.visit(this.path);
  }

  async openDashboard(): Promise<void> {
    await this.visit("/staff/dashboard");
  }

  /**
   * Navigates to buildings.
   */
  async goToBuildings(): Promise<void> {
    await this.visit("/staff/buildings");
  }

  /**
   * Navigates to customers.
   */
  async goToCustomers(): Promise<void> {
    await this.visit("/staff/customers");
  }

  /**
   * Navigates to contracts.
   */
  async goToContracts(): Promise<void> {
    await this.visit("/staff/contracts");
  }

  /**
   * Navigates to invoices.
   */
  async goToInvoices(): Promise<void> {
    await this.visit("/staff/invoices");
  }

  /**
   * Navigates to profile.
   */
  async goToProfile(): Promise<void> {
    await this.visit("/staff/profile");
  }
}
